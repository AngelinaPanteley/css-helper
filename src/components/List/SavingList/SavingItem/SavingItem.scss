@import '../../../../variables';

.Item {
  display: flex;
  height: 50px;
  align-items: center;
  color: $color-white;
  padding: 0 30px;
  justify-content: space-between;
  border-bottom: 2px solid $color-border;
  cursor: pointer;
  font-size: 20px;
  transition: all .1s ease-in-out;

  &:hover {
    background-color: $color-pink;
    transition: all .1s ease-in-out;
  }

  &:last-child {
    border: 0;
  }
}

.Button {
  border: none;
  background: none;
  outline: none;
  color: white;
  font-size: 22px;
  cursor: pointer;
  transition: all .1s ease-in-out;

  &:hover {
    text-shadow: 0 0 1px $color-white;
    transition: all .1s ease-in-out;
  }
}