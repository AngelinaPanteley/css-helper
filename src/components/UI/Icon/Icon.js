import React from 'react';

import styles from './Icon.scss';

const icon = (props) => {
  return (
    <i className={`${styles.icon} ${styles['icon_' + props.icon]}`} />
  )
}

export default icon;