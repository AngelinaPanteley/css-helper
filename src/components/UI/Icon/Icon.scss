@font-face {
  font-family: 'css-helper';
  src: url('../../../assets/icons/font/css-helper.eot?61525200');
  src: url('../../../assets/icons/font/css-helper.eot?61525200#iefix') format('embedded-opentype'),
       url('../../../assets/icons/font/css-helper.woff2?61525200') format('woff2'),
       url('../../../assets/icons/font/css-helper.woff?61525200') format('woff'),
       url('../../../assets/icons/font/css-helper.ttf?61525200') format('truetype'),
       url('../../../assets/icons/font/css-helper.svg?61525200#css-helper') format('svg');
  font-weight: normal;
  font-style: normal;
}
/* Chrome hack: SVG is rendered more smooth in Windozze. 100% magic, uncomment if you need it. */
/* Note, that will break hinting! In other OS-es font will be not as sharp as it could be */
/*
@media screen and (-webkit-min-device-pixel-ratio:0) {
  @font-face {
    font-family: 'css-helper';
    src: url('../font/css-helper.svg?61525200#css-helper') format('svg');
  }
}
*/
 
.icon:before {
  font-family: "css-helper";
  font-style: normal;
  font-weight: normal;
  speak: none;
 
  display: inline-block;
  text-decoration: inherit;
  width: 1em;
  margin-right: .2em;
  text-align: center;
  /* opacity: .8; */
 
  /* For safety - reset parent styles, that can break glyph codes*/
  font-variant: normal;
  text-transform: none;
 
  /* fix buttons height, for twitter bootstrap */
  line-height: 1em;
 
  /* Animation center compensation - margins should be symmetric */
  /* remove if not needed */
  margin-left: .2em;
 
  /* you can be more comfortable with increased icons size */
  /* font-size: 120%; */
 
  /* Font smoothing. That was taken from TWBS */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
 
  /* Uncomment for 3D effect */
  /* text-shadow: 1px 1px 1px rgba(127, 127, 127, 0.3); */
}
 
.icon_down:before { content: '\e800'; } /* '' */
.icon_left:before { content: '\e801'; } /* '' */
.icon_right:before { content: '\e802'; } /* '' */
.icon_up:before { content: '\e803'; } /* '' */
.icon_edit:before { content: '\e804'; } /* '' */
.icon_delete:before { content: '\e805'; } /* '' */
.icon_plus:before { content: '\e806'; } /* '' */
.icon_minus:before { content: '\e807'; } /* '' */
.icon_download:before { content: '\e808'; } /* '' */
.icon_upload:before { content: '\e809'; } /* '' */
.icon_menu:before { content: '\f008'; } /* '' */
.icon_copy:before { content: '\f0c5'; } /* '' */
.icon_account:before { content: '\f2be'; } /* '' */