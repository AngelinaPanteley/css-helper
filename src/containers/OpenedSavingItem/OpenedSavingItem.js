import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../../store/actions/index';
import styles from './OpenedSavingItem.scss';
import { calcStyles } from '../../shared/utility';
import Preview from '../../components/Preview/Preview';
import Pattern from '../../components/Code/Pattern/Pattern';

class OpenedSavingItem extends Component {
  constructor(props) {
    super(props);

    this.settings = this.props.settings[this.props.item.editorName];
    this.previewClass = this.settings.className;

    this.state = {
      styleValues: calcStyles(
        this.settings.style,
        props.item.controlValues,
        this.settings.controls
      ),
    }
  }

  openCopyHint = () => {
    this.props.openHint('Copied.');
  }

  render() {
    const wrapperStyles = [styles.Wrapper];
    if (this.props.isOpen) {
      wrapperStyles.push(styles.Open);
    } else {
      wrapperStyles.push(styles.Close);
    }

    let styleString = '';
    for (let styleKey in this.state.styleValues) {
      styleString = `${styleString}\n  ${styleKey}: ${this.state.styleValues[styleKey]};`;
    }
    const styleCode = `.${this.previewClass} {${styleString}\n}`;

    return (
      <div className={wrapperStyles.join(' ')}>
        <div className={styles.Pattern}>
          <Pattern code={styleCode}
            title='Styles'
            onCopy={this.openCopyHint}
          />
        </div>
        <div className={styles.Preview}>
          <Preview
            previewClass={this.settings.className}
            template={this.settings.template}
            styles={this.state.styleValues} />
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    settings: state.editors.settings,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    openHint: (text) => {
      dispatch(actions.openHint(text));
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(OpenedSavingItem);