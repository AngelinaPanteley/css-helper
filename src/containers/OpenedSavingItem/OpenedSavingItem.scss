@import '../../variables';

.Wrapper {
  @include flex-wrapper;

  border-bottom: 2px solid $color-border;
  padding: 10px 30px;
}

.Preview {
  width: 49%;
  display: flex;
  justify-content: center;
}

.Pattern {
  width: 49%;
  margin-bottom: -60px;
}