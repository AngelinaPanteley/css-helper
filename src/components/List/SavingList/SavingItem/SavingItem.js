import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './SavingItem.scss';
import Icon from '../../../UI/Icon/Icon';
import Auxiliary from '../../../../hoc/Auxiliary/Auxiliary';
import OpenedSavingItem from '../../../../containers/OpenedSavingItem/OpenedSavingItem';

class SavingItem extends PureComponent {
  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    onClick: PropTypes.func.isRequired,
    onEdit: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired,
    item: PropTypes.object.isRequired,
  }

  render() {
    const item = this.props.item;
    return (
      <Auxiliary>
        <div className={styles.Item}
          onClick={() => this.props.onClick(item.id)}>
          {item.title}
          <div className={styles.ButtonContainer}>
            <button
              onClick={(e) => {
                e.stopPropagation();
                this.props.onEdit(item.id);
              }}
              className={styles.Button}>
              <Icon icon='edit' />
            </button>
            <button
              onClick={(e) => {
                e.stopPropagation();
                this.props.onDelete(item.id);
              }}
              className={styles.Button}>
              <Icon icon='delete' />
            </button>
          </div>
        </div>
        <OpenedSavingItem item={item} isOpen={this.props.isOpen} />
      </Auxiliary>
    )
  }
}

export default SavingItem;