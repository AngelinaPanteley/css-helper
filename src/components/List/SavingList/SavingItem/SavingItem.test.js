"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import { MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import SavingItem from './SavingItem.js';
import reducer from '../../../../store/reducers/index';
import Icon from '../../../UI/Icon/Icon.js';
import OpenedSavingItem from '../../../../containers/OpenedSavingItem/OpenedSavingItem.js';

configure({ adapter: new Adapter() });

const store = createStore(reducer);

describe('>>>SavingItem', () => {

  //SNAPSHOTS

  const props = {
    isOpen: true,
    item: {
      id: '12345',
      title: 'title',
    },
    onClick: () => { },
    onEdit: () => { },
    onDelete: () => { },
  };

  const itemComponent = <Provider store={store}><MemoryRouter>
    <SavingItem {...props} />
  </MemoryRouter></Provider>;

  const component = renderer.create(itemComponent);

  it('renders correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;
  const shallowComponent = <SavingItem {...props} />;

  beforeEach(() => {
    wrapper = shallow(shallowComponent);
  });

  it('renders two icons', () => {
    chaiExpect(wrapper.find(Icon)).to.have.lengthOf(2);
  });
  it('renders opened saving item if isOpen=true', () => {
    chaiExpect(wrapper.find(OpenedSavingItem)).to.have.lengthOf(1);
  });
});
