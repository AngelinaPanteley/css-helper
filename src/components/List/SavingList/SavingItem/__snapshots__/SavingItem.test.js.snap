// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>SavingItem renders correctly 1`] = `
Array [
  <div
    onClick={[Function]}
  >
    title
    <div>
      <button
        onClick={[Function]}
      >
        <i
          className="undefined undefined"
        />
      </button>
      <button
        onClick={[Function]}
      >
        <i
          className="undefined undefined"
        />
      </button>
    </div>
  </div>,
  <div
    className=" "
  >
    <div>
      <div>
        <p>
          <span>
            Styles
          </span>
          <button
            onClick={[Function]}
          >
            <i
              className="undefined undefined"
            />
          </button>
        </p>
        <pre>
          <code>
            .undefined {
}
          </code>
        </pre>
      </div>
    </div>
    <div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </div>,
]
`;
